Enable strongly typed ops for deployment

mcremon-meta opened the pull request:
Summary:
As titled. This allows a severe reduction in codesize by only using the bare minimum cpp code. Right now, this diff only implements that option for quantized fully connected per tensor.
This diff is introducing opt level 4 to use for deployment purposes. The idea is similar to -Os in most compilers, we just use 4 for simplicity and because for now, everything should be inclusive by construction. This decision can be re-visited later!

Differential Revision: D79867630

msluszniak commented on backends/cadence/hifi/operators/op_quantized_fully_connected_asym8sxasym8s_asym8s_per_tensor_out.cpp:
```diff
@@ -0,0 +1,65 @@
+#include <executorch/backends/cadence/hifi/kernels/kernels.h>
+#include <executorch/runtime/kernel/kernel_includes.h>
+
+namespace cadence {
```
Nested namespaces with `::` are prefered since c++17, https://clang.llvm.org/extra/clang-tidy/checks/modernize/concat-nested-namespaces.html

mcremon-meta committed:
Enable type-specific ops for deployment (#13230)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/13230

As titled. This allows a severe reduction in codesize by only using the bare minimum cpp code. Right now, this diff only implements that option for quantized fully connected per tensor.
This diff is introducing opt level 4 to use for deployment purposes. The idea is similar to -Os in most compilers, we just use 4 for simplicity and because for now, everything should be inclusive by construction. This decision can be re-visited later!

Reviewed By: zonglinpeng

Differential Revision: D79867630
